frontends/web/src/routes/account/walletconnect/components/session-card/session-card.module.css

```diff
@@ -0,0 +1,99 @@
+.container {
+    background-color: var(--background-session-card);
+    display: flex;
+    width: 100%;
+    padding: calc(var(--space-half) + var(--space-quarter));
+}
+
+.container p {
+    color: var(--color-default);
+    font-size: 16px;
+    margin: 0;
+}
+
+.container .textDataContainer {
+    display: flex;
+    flex-direction: column;
+    gap: calc(var(--space-quarter) + var(--space-half));
+}
+
+.container .textDataContainer .receiveAddress {
+    color: var(--color-secondary);
+}
+
+.container .textDataContainer .accountName {
+    font-weight: bold;
+    width: 240px;
+    padding-right: 16px;
+}
+
+.container .dAppMetadataAndIconContainer {
+    align-items: center;
+    display: flex;
+}
+
+.container .dAppMetadataAndIconContainer img {
+    width: var(--item-height);
+    height: var(--item-height);
+    margin-left: calc(var(--space-quarter) + var(--space-half));
+}
+
+.container .dAppNameAndUrlContainer {
+    display: flex;
+    flex-direction: column;
+    gap: calc(var(--space-quarter) * 2);
+    width: 240px;
+}
+
+.container .accountNameAndWalletContainer {
+    display: flex;
+}
+
+.container .buttonAndIconContainer {
+    align-items: flex-end;
+    display: flex;
+    margin-left: auto;
+}
+
+.container .buttonAndIconContainer button {
+    height: unset;
+    padding: calc(var(--space-eight) + var(--space-quarter)) calc(var(--space-default) + var(--space-default));
+}
+
+@media (max-width: 768px) { 
+    .container {
+        padding: calc(var(--space-quarter) * 2);
+    }
+
+    .container p { 
+        font-size: var(--size-default);
+    }
+
+    .container .textDataContainer {
+        gap: var(--space-quarter);
+        width: 50%;
+    }
+
+    .container .dAppNameAndUrlContainer,
+    .container .textDataContainer .accountName {
+        width: 100%;
+    }
+
+    .container .buttonAndIconContainer {
+        flex-direction: column;
+        justify-content: flex-end;
+    }
+
+    .container .buttonAndIconContainer button {
+        font-size: var(--size-default);
+        height: var(--item-height-small);
+        padding: var(--space-eight) calc(var(--space-quarter) + var(--space-half));
+    }
+
+    .container .buttonAndIconContainer img {
+        height: var(--item-height-small);
+        width: var(--item-height-small);
+        margin-top: var(--space-default);
+        margin-bottom: var(--space-default);
+    }
```

Comment:
Is the additional CSS specifity needed or could you remove all `.container` classes except for the container styles itself?

i.e. `.container .buttonAndIconContainer button` -> `.buttonAndIconContainer button` ?

Reply:
actually in the example of button I think it would be better to just have a class to pass to `<Button className={style.buttonDisconnect}` so that the CSS doesn't depend on the child components DOM.

i.e. in theory if we ever change Button component internally depending on `.buttonAndIconContainer button` could break at some point and it would be hard to notice.

Reply:
Thanks for these comments. I was unsure how we wanna tackle this. I was trying to reduce the usage of classnames and rather target it semantically. 

But your comment regarding the breaking when we restructure divs is valid - I guess this is a  "give or take" situation. I'll reduce the complexity by just resorting to classnames as you suggested.